src/plugins/intel_gna/backend/gna_limitations.cpp

```diff
@@ -121,52 +121,100 @@ std::string RectLimitByChannelsAndPrecision::GetErrorOrEmpty(const uint32_t h, c
+    const RangeMultipleLimit kInputChannelsNumberLimit{{8, 384, "number of input channels"}, 8};
+
+    const RangeMultipleLimit kKernelNumberLimit{{8, 1024, "number of kernels"}, 8};
+    const RectLimitByChannelsAndPrecision kKernelLimit{
+        {{{96, {7, 7}}, {136, {7, 5}}, {168, {7, 4}}, {240, {7, 3}}, {384, {7, 2}}}},
+        {{{48, {7, 7}}, {64, {7, 5}}, {80, {7, 4}}, {120, {7, 3}}, {384, {7, 1}}}},
+    };
+    const auto& kStrideLimit = kKernelLimit;
+    const RangeLimit2D dilationLimit{{convDilationHeight, convDilationHeight, "dilation height"},
+                                     {convDilationWidth, convDilationWidth, "dilation width"}};
+    auto error = kInputHWLimit.GetErrorOrEmpty(inHeight, inWidth);
+
+    error += kKernelNumberLimit.GetErrorOrEmpty(kernelN);
+    error += kInputChannelsNumberLimit.GetErrorOrEmpty(inChannels);
+    error += kKernelLimit.GetErrorOrEmpty(kernelH, kernelW, inPrecision, inChannels, "kernel");
+    error += kStrideLimit.GetErrorOrEmpty(strideH, strideW, inPrecision, inChannels, "convolution stride");
+
+    const RangeLimit kKernelStrideHLimit{1, kernelH, "kernel stride height (must be up to kernel height)"};
+    const RangeLimit kKernelStrideWLimit{1, kernelW, "kernel stride width (must be up to kernel width)"};
+
+    error += kKernelStrideHLimit.GetErrorOrEmpty(strideH);
+    error += kKernelStrideWLimit.GetErrorOrEmpty(strideW);
 
     error += dilationLimit.GetErrorOrEmpty(dilationH, dilationW);
 
-    if (exception)
-        ThrowIfNotEmpty(prefix, error);
-
-    return error.empty() ? true : false;
+    return ValidationSuccesful(throwOnError, error, name, "Convolution2D");
 }
 
 bool Validator_30::ValidatePooling2D(const std::string& name,
     const uint32_t windowH, const uint32_t windowW,
     const uint32_t strideH, const uint32_t strideW,
-    bool exception) const {
-    const std::string prefix = "Layer Pooling2D: " + name + ":";
-
-    auto error = poolingWindowLimit.GetErrorOrEmpty(windowH, windowW, "pooling window");
+    const bool throwOnError) const {
+    const VectorOrSquareLimit kPoolingWindowLimit{3, 1, 1};
+    auto error = kPoolingWindowLimit.GetErrorOrEmpty(windowH, windowW, "pooling window");
     const RangeLimit poolingStrideHLimit{ 1, windowH, "pooling stride height (must be up to pooling window height)" };
     const RangeLimit poolingStrideWLimit{ 1, windowW, "pooling stride width (must be up to pooling window width)" };
 
     error += poolingStrideHLimit.GetErrorOrEmpty(strideH);
     error += poolingStrideWLimit.GetErrorOrEmpty(strideW);
 
-    if (exception)
-        ThrowIfNotEmpty(prefix, error);
+    return ValidationSuccesful(throwOnError, error, name, "Pooling2D");
+}
+
+bool Validator_30::IsPaddingSupported() const {
+    return false;
+}
+
+bool Validator_35::ValidateCnn2D(const std::string& name, const uint32_t inHeight, const uint32_t inWidth,
+    const uint32_t inChannels, const uint32_t kernelH, const uint32_t kernelW, const uint32_t kernelN,
```

Comment:
previously all this values were initialized once when validator was created
currently this will be instantiated each time method will be invoked.

I see that this values are predefined.
Wouldn't be better to put this *limit objects as static const/constexpr class members?

Reply:
changed to `static const` class members